src/easylink/graph_components.py

```diff
@@ -42,63 +48,38 @@ class InputSlot:
     this ``InputSlot``. If the data is invalid, the function should raise an exception 
     with a descriptive error message which will then be reported to the user.
     **Note that the function *must* be defined in the** :mod:`easylink.utilities.validation_utils` 
     **module!**"""
-    splitter: Callable[[list[str], str, Any], None] | None = field(
-        default=None, compare=False
-    )
-    """A function that splits the incoming data to this ``InputSlot`` into smaller
-    pieces. The primary purpose of this functionality is to run sections of the 
-    pipeline in an embarrassingly parallel manner. **Note that the function *must* 
-    be defined in the **:mod:`easylink.utilities.splitter_utils`** module!**"""
-    splitter_origin_node: str = None
-    """The name of the node where a splitter is originally defined."""
-    splitter_origin_slot: str = None
-    """The name of the ``InputSlot`` where a splitter is originally defined."""
 
     def __eq__(self, other: Any) -> bool | NotImplementedType:
-        """Checks if two ``InputSlots`` are equal.
-
-        Two ``InputSlots`` are considered equal if their names, ``env_vars``, and
-        names of their ``validators`` and ``splitters`` are all the same.
-        """
+        """Checks if two ``InputSlots`` are equal."""
         if not isinstance(other, InputSlot):
             return NotImplemented
-        splitter_name = self.splitter.__name__ if self.splitter else None
-        other_splitter_name = other.splitter.__name__ if other.splitter else None
+        validator_name = self.validator.__name__ if self.validator else None
+        other_validator_name = other.validator.__name__ if other.validator else None
         return (
             self.name == other.name
             and self.env_var == other.env_var
-            and self.validator.__name__ == other.validator.__name__
-            and splitter_name == other_splitter_name
-            and self.splitter_origin_node == other.splitter_origin_node
-            and self.splitter_origin_slot == other.splitter_origin_slot
+            and validator_name == other_validator_name
         )
 
     def __hash__(self) -> int:
-        """Hashes an ``InputSlot``.
-
-        The hash is based on the name of the ``InputSlot``, its ``env_var``, and
-        the names of its ``validator`` and ``splitter``.
-        """
-        splitter_name = self.splitter.__name__ if self.splitter else None
+        """Hashes an ``InputSlot``."""
+        validator_name = self.validator.__name__ if self.validator else None
         return hash(
             (
                 self.name,
                 self.env_var,
-                self.validator.__name__,
-                splitter_name,
-                self.splitter_origin_node,
-                self.splitter_origin_slot,
+                validator_name,
             )
         )
 
 
-@dataclass()
+@dataclass(frozen=True)
```

Comment:
not precisely necessary but figured we should freeze if possible? We no longer modify output slots with this new approach

Reply:
Not sure I understand - we can't freeze input slots b/c I do indeed change them. What would the TODO be?

Reply:
Ah, good catch. `eq=True` is the default anyway. I prewviously had to add __eq__ and __hash__ b/c I had to special-handle callables which no longer exist in OutputSlots (still need to do that for InputSlots, though)

Reply:
> we can't freeze input slots b/c I do indeed change them

Interesting, maybe the TODO is to stop doing that so we can freeze them? But I haven't followed this closely so feel free to disregard.

Reply:
> callables which no longer exist in OutputSlots (still need to do that for InputSlots, though)

We really should move toward passing function _names_ rather than callables, as that's what we end up using anyway (we can't jam an actual callable into a Snakefile).